### PR TITLE
 NXP-21179: add nuxeo-sardine

### PR DESCRIPTION
**Pull request by Gabez0r**

### CODE REVIEW

**Comment by guirenard** [comment]
:+1:

File: elements/nuxeo-sardine.html [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+              id: 'sardine',
+              icon: 'images/gift.png',
+              label: 'I love sardines!!! 🐟🐟🐟💝💝💝',
+            },
+            run: this._sardineOn.bind(this),
+          });
+        }.bind(this));
+      },
+
+      _sardineOn: function() {
```
Comment (nelsonsilva) [inline elements/nuxeo-sardine.html]: nit: if this in `nuxeo-sardine` and the only thing we turn on / off is the sardine perhaps `_on`, '_off`, 
`animate` would be enough 🤷‍♂️

File: elements/nuxeo-sardine.html [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        this._onkeypress = this._onkeypress || function(e) {
```
Comment (nelsonsilva) [inline elements/nuxeo-sardine.html]: why not move these handlers to actual methods `_onkeypress: function(e) {...}` ?

File: elements/nuxeo-sardine.html [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+          window.addEventListener('mouseout', this._onpointerout, false);
+        }
+        window.addEventListener('keyup', this._onkeypress, false);
+      },
+
+      _animateSardine: function() {
+        if (this._idleTimeout) {
+          clearTimeout(this._idleTimeout);
+        }
+        this._idleTimeout = setTimeout(function() {
```
Comment (nelsonsilva) [inline elements/nuxeo-sardine.html]: seems weird to register a timeout in the request animation callback...

File: elements/nuxeo-sardine.html (hunk 2 above)
Comment (Gabez0r) [reply elements/nuxeo-sardine.html]: Just to define them as lazily as possible.

**Comment by nmpcunha** [comment]
👍

**Pull request merged**